### ISSUE TITLE
No sound recorded

### ISSUE THREAD

**Issue by PhilippeMoussa** [theevann/streamlit-audiorecorder#1]
Hey,
This component is exactly what I'm looking for so, thanks!
But, I can't make it work. 
I tried the following on my Mac:

audio = audiorecorder("Click to record", "Recording...")
if len(audio) > 0:
    st.audio(audio, sample_rate = 16000)

I am required to specify the sample_rate.
The app does require access to my microphone - good.
And whatever rate I try, I get only some white noise when trying to play the record.

Anything I'm not doing right? Thanks in advance!

**Comment by theevann** [comment]
Hello,
Thank you for using it, and thank you for reporting the issue :)

Ok, I took some time to check this out, and it seems to be only appearing on the latest streamlit release.
Specifically, it is linked to this [pull request](https://github.com/streamlit/streamlit/pull/5554/).

They are doing some fancy things with the numpy array and apparently messing things up for us.
A working workaround I found is to convert the numpy array to bytes before feeding it to `st.audio`.
For now, just do:
```
st.audio(audio.tobytes())
```
And everything should just work as expected :)

I'll try to investigate more to see if I can preprocess the array to make is `st.audio` friendly, otherwise I'll just update the doc.
